### DISCUSSION TITLE
Installation source file missing

### DISCUSSION THREAD

**Discussion by dbr81** [nevarok/NevarokML#5] [Q&A] [answered]
Hello, I tried to install the plugin from the GitHub, but I'm stuck on the installation, the source folder shows an error, but I cant unzip the exe file as per the install video. the source file is not a file  can extract. 

thank you

**Comment by nevarok** [comment]
Hello @dbr81, I hope you're doing well. I noticed you're encountering some challenges with installing a plugin from GitHub. I'd be more than happy to assist you with this. Could you kindly provide some additional details about the steps you followed during the installation process? Specifically, did you follow the instructions in the video located at https://www.youtube.com/watch?v=nFbgVGCifA8? If yes, could you please specify at which stage you encountered the error?

It's important to note that when installing from GitHub, there can be an issue where a warning message appears in the settings even if the installation is successful. This occurs due to Git LFS, as zip file is partitioned into multiple segments. Despite the warning message, the plugin may still function correctly. Your clarification on these points will help me provide you with more accurate guidance.

**Comment by dbr81** [comment]
Hi, thank you for your quick response,

yes is was following the steps in the video and I'm stuck at the backend
process.  In the editor Preferences i have the error: archived not found,
and the option available don't resolve the issue.
i did verify if the plug in still worked, but there's no files in the
content Drawer.

[image: image.png]
thank you

if the zip file is partitioned, is there a way to reconcile them?

**Reply by nevarok** [reply]
Hello @dbr81,

Thank you for providing more details about the issue you're encountering with the plugin installation. Let's try to resolve it step by step:

1) Plugin Content Display: Please check if the following setting is enabled in your project, as it's crucial for ensuring the plugin's content folders are displayed correctly. Visit [this link](https://github.com/nevarok/NevarokML/assets/6491805/00927ec9-9243-46a7-b989-172a333b33a8) to compare with your current settings and make sure they match.

2) Unreal Engine Version: Compatibility is key. Could you confirm that you're using Unreal Engine 5.2? This specific version is necessary for the proper functioning of the plugin.

3) File Location Verification: Make sure that nevarok_ml.exe and all other files from the archive are located in the directory /Plugins/NevarokML/Source/NevarokMLExe. The correct file placement is essential for the plugin to function properly.

4) Extracting the Files: If the files are not yet extracted, please use 7zip or a similar tool to extract NevarokMLExe.zip located in /Plugins/NevarokML/Source. You can do this by right-clicking on the archive and selecting 7-Zip -> Extract Here. This should help in assembling the partitioned files correctly.

I hope these steps assist you in moving forward with the installation. Please don't hesitate to reach out if you have any further questions or require additional assistance.

**Comment by dbr81** [comment]
hi,

1. the point 1 helps to show the maps, thank you
2. i confirm is UE 5.2
3. i fixed the location of the files
4. after extracting the files, i don't have any option to extract the .exe
file

this is what i have
[image: image.png]
thank you

**Reply by nevarok** [reply]
Once you have successfully extracted the files, you should be all set to proceed. Here's what to do next:

1) Launch an Example: Try launching any of the examples located within the plugin's content folder. This will help you verify if the installation has been successful and the plugin is functioning as expected.

2) Automatic Execution of the Executable: Keep in mind that the nevarok_ml.exe should automatically launch as soon as the training starts within the plugin. This is a good indicator that the plugin is operational.

3) Further Assistance: If you find that the above steps do not resolve your issue or if you encounter any additional challenges, please don't hesitate to reach out. We can set up a call, and I will guide you through the installation process step by step to ensure everything is set up correctly.

I'm here to help, so feel free to contact me if you need further assistance or clarification. Good luck with your installation, and I look forward to hearing about your successful setup!

**Comment by dbr81** [comment]
Thank you again for your support on this.
Does the files need to be absolutely saved in the engine root or can they
be saved in the project only?

thank you for the clarification

**Reply by nevarok** [reply]
there are 2 options where to extract plugin files

1) extract into [Project Folder]/Plugins/NevarokML (example  D:\repo\SomeUEProject\Plugins\NevarokML)
2) extract into UE_5.2/Engine/Plugins/Marketplace/NevarokML (example C:\Program Files\Epic Games\UE_5.2\Engine\Plugins\Marketplace\NevarokML)


Regarding backend archive

1) zip archive can be extracted in the folder where it's located Plugins/NevarokML/Source ![example image](https://github.com/nevarok/NevarokML/assets/6491805/c935af21-b2e8-418d-8d59-05d437241948)
2) there is also a possibility to change the executable location, please refer to the ![image](https://github.com/nevarok/NevarokML/assets/6491805/4ab23fd9-ed51-47d6-b077-2d7e88180d9c)

**Comment by dbr81** [comment]
the issue is that i can't find the or extract the .exe, all the rest works
except that

This means I can see the map test, read the c++ script and blueprint but I
can't run the test.

I'm trying to understand the process from a to z because I'm trying to
build on for a game specific and all the documentation I can find is not
clear or incomplete.  Your plugin was the best solution to learn.

thank you

**Reply by nevarok** [reply, marked as answer]
Could you please check if all of these files are present in the plugin source folder when you do the checkout https://github.com/nevarok/NevarokML/tree/main/UE_5.2/NevarokML/Source.
![image](https://github.com/nevarok/NevarokML/assets/6491805/34e43a79-6ee4-45e3-a9cd-5f823766cace)

1) if the files are absent for some reason you can manually download them (https://github.com/nevarok/NevarokML/tree/main/UE_5.2/NevarokML/Source [NevarokMLExe.zip.001](https://github.com/nevarok/NevarokML/blob/main/UE_5.2/NevarokML/Source/NevarokMLExe.zip.001), [NevarokMLExe.zip.002](https://github.com/nevarok/NevarokML/blob/main/UE_5.2/NevarokML/Source/NevarokMLExe.zip.002), [NevarokMLExe.zip.003](https://github.com/nevarok/NevarokML/blob/main/UE_5.2/NevarokML/Source/NevarokMLExe.zip.003)) from the repo or try to make a clean checkout of the repo again.

2) if the files are present you should perform extraction:
![image](https://github.com/nevarok/NevarokML/assets/6491805/cbd4ee3e-9491-42a3-90b5-f231363c0e91)
![image](https://github.com/nevarok/NevarokML/assets/6491805/01e15d82-fb08-4a9a-8a72-7ff71bfcdc78)
![image](https://github.com/nevarok/NevarokML/assets/6491805/feeddf19-d314-4458-82e4-b36a8bcd205c)
![image](https://github.com/nevarok/NevarokML/assets/6491805/00af7b45-f2f9-4ad8-8ddb-a7f5640007cf)

**Comment by dbr81** [comment]
thank you very much,  it seems that the extraction tool i have was not
recognizing the files, i use 7zip and that fixes the issue.

I would like to ask a question to understand how it works?
Do you have any suggestions for books or sites? If I want to create my own
RL tool for a fighting game, I would like the AI to learn how to play to
test the game.

thank you very much

**Reply by nevarok** [reply]
Hello @dbr81,

I'm truly delighted to hear that the solution worked out for you!

When it comes to diving into the learning process, I always advocate for starting with the fundamentals and progressively building up your confidence. In my experience, approaching learning with a specific problem or project in mind can be incredibly effective and rewarding.

As for book recommendations, it's a bit challenging to suggest specific titles without knowing your current level of understanding in Machine Learning. The field is vast, and there are countless resources available, each tailored to different levels of expertise.

However, I would definitely recommend beginning with the [Deep Reinforcement Learning course by Hugging Face](https://huggingface.co/learn/deep-rl-course/unit0/introduction). This course provides a comprehensive introduction and could be an excellent starting point for your journey. Additionally, for a more interactive and engaging learning experience, you might find Nicholas Renotte's YouTube channel extremely helpful. He offers some clear and accessible explanations that are ideal for beginners in Reinforcement Learning and Machine Learning. You can explore his content [here](https://www.youtube.com/@NicholasRenotte).

Best of luck on your learning journey, and if you have any more questions or need further guidance, please don't hesitate to reach out!

**Comment by dbr81** [comment]
Thanks for the tip!
